### ISSUE TITLE
Add a method to Servo to clean the history

### ISSUE THREAD

**Issue by paulrouget** [servo/servo#19284]
A `PurgeHistory(BrowserId)` event.

This would clear the history.

On completion, we would get a `HistoryChanged` event.

/cc @asajeffrey what would it take to implement that?

**Comment by cbrewster** [comment]
By purging do you mean removing all history entries in the past and future?

**Comment by paulrouget** [comment]
Yes.

**Comment by asajeffrey** [comment]
It probably wouldn't be too tricky to implement, but we'd need a compelling argument for why we should have some non-standard behaviour. @paulrouget what's a use-case?

**Comment by paulrouget** [comment]
> what's a use-case?

That's a feature found in some browsers. A menu to clear history. It's also a method found in Webkit and Gecko.

In Firefox, clearing history make it so we can't go back or forward anymore, but the current document is conserved.

**Comment by asajeffrey** [comment]
Fair enough. Is any of this spec'd somewhere? Is the `HistoryChanged` event? It looks like `PopState` is the one specified in https://html.spec.whatwg.org/multipage/browsing-the-web.html#history-traversal:popstateevent, though that's not a great name.

**Comment by paulrouget** [comment]
I don't think there is any spec. The `HistoryChanged` event is already here and will like be fired automatically. To be clear, I am not talking about the browserapi. We just need another `WindowEvent`.

@asajeffrey this is not urgent, and once I need it, I will implement it myself. I just wanted to know if there was any difficulties in implementing this beside finding all the non active pipelines and destroying them.
